exist-core/src/main/java/org/exist/xquery/functions/fn/FunBaseURI.java

```diff
@@ -160,10 +161,12 @@ else if ((type == Node.PROCESSING_INSTRUCTION_NODE ||
                 URI relativeURI = null;
                 URI baseURI = null;
                 try {
-                    final String uri = domNode.getBaseURI();
-                    if (uri != null) { 
-                        relativeURI = new URI(uri);
-                        baseURI = new URI(context.getBaseURI() + "/");
+                    if (domNode != null) {
+                        final String uri = domNode.getBaseURI();
+                        if (StringUtils.isNotBlank(uri)) {
+                            relativeURI = new URI(uri);
+                            baseURI = new URI(context.getBaseURI() + "/");
```

Comment:
I am just wondering if we also have to check here `if(context!=null && context.isBaseURIDeclared())` like in your change to DocumentImpl above?

Reply:
not sure... makes probably sense

Reply:
it makes sense to check for a context `null` check. If not null, it looks that in the context at least a value is set for the baseUri.

- if context is null, how to handle?
- interesting comments in the xqueryContext method, not sure if the current code is OK, but it might be implementation dependant. @ljo 's remark makes sense?

<img width="739" alt="image" src="https://user-images.githubusercontent.com/1700062/104643293-7d363400-56ac-11eb-81fd-efb2a90a3ffa.png">